File: 2024/src/01/day01a.cc
#include <bits/stdc++.h>
using namespace std;

int main()
{
    vector<int> lefties;
    vector<int> righties;

    // Read input
    ifstream file{"../input/day01.txt"};
    string line;
    while (getline(file, line))
    {
        int a, b;
        stringstream ss(line);
        ss >> a >> b;
        lefties.push_back(a);
        righties.push_back(b);
    }

    // Process input
    sort(lefties.begin(), lefties.end());
    sort(righties.begin(), righties.end());
    int result = inner_product(lefties.begin(), lefties.end(), righties.begin(), 0, plus<>(), [](int l, int r)
                               { return abs(l - r); });
    cout << result;
    return 0;
}

File: 2024/src/01/day01a.js
const fs = require("fs");

lefties = [];
righties = [];
// Open the file and read it line by line
fs.readFile("input/day01.txt", "utf8", (err, data) => {
  if (err) throw err;
  const lines = data.split("\n");
  lines.forEach((line) => {
    let parts = line.trim().split();
    let [l, r] = parts[0].split("  ").map(Number);
    lefties.push(l);
    righties.push(r);
  });

  // Process
  let result = 0;
  lefties.sort();
  righties.sort();
  for (let i = 0; i < lefties.length; i++) {
    result += Math.abs(lefties[i] - righties[i]);
  }
  console.log(result);
});
